### ISSUE TITLE
WARNING: Plugin Definition Error when running RSpec

### ISSUE THREAD

**Issue by lrnzdvs** [burtlo/chefspec-ohai#1]
```
$ chef exec rspec spec/ohai_plugins/apache_modules_spec.rb
.[2016-12-02T14:18:59-06:00] WARN: Plugin Definition Error: <files/default>: collect_data already defined on platform default
.

Finished in 0.20469 seconds (files took 1.54 seconds to load)
2 examples, 0 failures
```

Ohai plugin possibly evaluated twice.

**Comment by burtlo** [comment]
Thank you. I'll have to track that down.

**Comment by mhedgpeth** [comment]
I'm seeing this as well.

**Comment by JaWe09** [comment]
me too.

**Comment by haidangwa** [comment]
@burtlo any progress on this? I see this issue has been around a while, and I just encountered it today during my testing.

**Comment by burtlo** [comment]
I never did track it down. Oh my it has been a long time. Let me add it to my calendar.

**Comment by burtlo** [comment]
Alright, the issue appears only when you more than one expectation. That is because the helper `let(:plugin_source)` is loaded each time there is an RSpec example.

During the `before(:each)` I use the Ohai Loader with a stubbed controller and the plugin source to load the file. I assume that the loader would be cleared each time, as I making a new one each time. But I'm beginning to suspect that the Ohai plugin class is loaded and then stays resident in memory between the examples.

I'm going to look into Ohai to see if there is a way to clear all the ohai plugins. If one does not exist I will need to watch for new constants that are created (a new class - this new Ohai plugin), track that, and then undefine this constant after each spec is complete.

**Comment by burtlo** [comment]
For each example the plugin file is loaded. When its loaded the first
time Ohai uses the DSL to create a class object that it assigns to a
constant and stores in a particular module (`Ohai::NamedPlugin`). This is usually alright but
does create a problem when you attempt to load this same class definition
over and over again.

To get around that I added to the before/after to capture the current
constants that are defined and remove them after each run. This means
that every specification will now reload the ohai plugin class.

**Comment by burtlo** [comment]
Thanks for your patience with regard to this issue. It should be fixed in the `0.1.1` release.
